platform/branding-registry-impl/src/main/java/org/codice/ddf/branding/impl/BrandingRegistryImpl.java

```diff
@@ -14,19 +14,58 @@
 package org.codice.ddf.branding.impl;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
+import java.util.TreeMap;
 import org.apache.commons.lang3.StringUtils;
 import org.codice.ddf.branding.BrandingPlugin;
 import org.codice.ddf.branding.BrandingRegistry;
+import org.codice.ddf.platform.util.ServiceComparator;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceListener;
+import org.osgi.framework.ServiceReference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class BrandingRegistryImpl implements BrandingRegistry {
   private static final Logger LOGGER = LoggerFactory.getLogger(BrandingRegistryImpl.class);
 
-  private List<BrandingPlugin> brandingPlugins = Collections.emptyList();
+  private Map<ServiceReference, BrandingPlugin> brandingPlugins =
+      Collections.synchronizedMap(
+          new TreeMap<>(
+              new ServiceComparator() {
+                public int compare(ServiceReference ref1, ServiceReference ref2) {
+                  return ref2.compareTo(ref1);
+                }
+              }));
+
+  private BrandingListener brandingListener = new BrandingListener();
+
+  public void init() {
+    BundleContext context = getContext();
+    context.addServiceListener(brandingListener);
```

Comment:
If you provide a filter here, i.e.
```
context.addServiceListener(brandingListener, "(objectclass=" + BrandingPlugin.class.getName() + ")");
```
You'll only be notified when a BrandingPlugin service changes